v4/source/jobs.yaml

```diff
@@ -0,0 +1,171 @@
+  '/jobs':
+    get:
+      tags:
+        - jobs
+      summary: Get the jobs.
+      description: |
+        Get a page of jobs. Use the query parameters to modify the behaviour of this endpoint.
+        __Minimum server version: 4.1__
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - name: page
+          in: query
+          description: The page to select.
+          default: "0"
+          type: string
+        - name: per_page
+          in: query
+          description: The number of jobs per page.
+          default: "60"
+          type: string
+      responses:
+        '200':
+          description: Job list retrieval successful
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Job'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+
+    post:
+      tags:
+      - jobs
+      summary: Create a new job.
+      description: |
+        Create a new job.
+        __Minimum server version: 4.1__
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - in: body
+          name: body
+          description: Job object to be created
+          required: true
+          schema:
+            type: object
+            required:
+              - type
+            properties:
+              type:
+                type: string
+                description: The type of job to create
+              data:
+                type: object
+                description: An object containing any additional data required for this job type
+      responses:
+        '201':
+          description: Job creation successful
+          schema:
+            $ref: '#/definitions/Job'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+
+  '/jobs/{job_id}':
+    get:
+      tags:
+        - jobs
+      summary: Get a job.
+      description: |
+        Gets a single job.
+        __Minimum server version: 4.1__
+        ##### Permissions
+        Must have `manage_system` permission.
+      parameters:
+        - name: job_id
+          in: path
+          description: Job GUID
+          required: true
+          type: string
+      responses:
+        '200':
+          description: Job retrieval successful
+          schema:
+            $ref: '#/definitions/Job'
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+        '403':
+          $ref: '#/responses/Forbidden'
+        '404':
+          $ref: '#/responses/NotFound'
+
+  '/jobs/{job_id}/cancel':
+    get:
```

Comment:
Does this delete the job? Or is cancel a separate action? If it's performing a delete we should use DELETE, otherwise we should use POST (we use that for actions not covered by the HTTP verbs we're using)